services/navigation/builtin/explore.go

```diff
@@ -2,9 +2,46 @@ package builtin
 
 import (
 	"context"
+	"math"
+	"math/rand"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/utils"
+
+	frame "go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
 )
 
-//nolint:unparam
+const defaultDistanceMM = 100 * 1000
+
 func (svc *builtIn) startExploreMode(ctx context.Context) {
 	svc.logger.Debug("startExploreMode called")
+
+	svc.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer svc.activeBackgroundWorkers.Done()
+
+		// Send motionCfg parameters through extra until motionCfg can be added to Move()
+		extra := map[string]interface{}{"motionCfg": svc.motionCfg}
+
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+
+			//nolint:gosec
+			randAngle := (rand.Float64() - 0.5) * math.Pi
+			destination := frame.NewPoseInFrame(svc.base.Name().Name, spatialmath.NewPose(
+				r3.Vector{
+					X: defaultDistanceMM * math.Sin(randAngle),
+					Y: defaultDistanceMM * math.Sin(randAngle),
```

Comment:
X and Y will always be equal here. You will sample destinations along a straight diagonal line.

The use of `Sin` will also mean your points will tend to cluster towards the extremities of your mappable area rather than being evenly distributed. Recommend not doing that.
Consider:
```
r3.Vector{
  X: defaultDistanceMM * (2*rand.Float64() - 1.0),
  Y: defaultDistanceMM * (2*rand.Float64() - 1.0),
  Z: 0.
}
```